src/components/GraphiQL.js

```diff
@@ -475,7 +475,13 @@ export class GraphiQL extends React.Component {
 
   _fetchQuery(query, variables, operationName, cb) {
     const fetcher = this.props.fetcher;
-    const fetch = fetcher({ query, variables, operationName });
+    const parsedVariables =
+      variables && variables.trim() !== '' ? JSON.parse(variables) : null;
```

Comment:
It's a good point. I guarded from an empty strings here, but it's possible that input is malformed. In this case it would be an error in the console and no request would be sent (a bit confusing to the user).

I agree, catching exception would be more robust error handling here. Though I would suggest to always set `jsonVariables` to `null`: 

```javascript
jsonVariables = null;
```

because of the 2 reasons:

* At this point we already know that JSON is either malformed  or empty, so it doesn't make sense to send it to the server
* In a long term it would be nice to require server to support only JSON object for variables and not both: string and JSON object (which is the case at the moment - most server implementations support string variables only to be compatible with GraphiQL, for all other requests sources JSON object is expected)